src/main/webapp/linkWizard/SelectSampleConsent.js

```diff
@@ -187,6 +192,27 @@ export const SelectSampleConsent = hh(class SelectSampleConsent extends Componen
     return sampleCollectionList;
   };
 
+  handleCheck = () => {
+    this.setState(prev => {
+      prev.onGoingProcess = !this.state.onGoingProcess;
+      prev.endDate = null;
+      return prev;
+    }, () => this.props.updateDateRange(this.state.onGoingProcess, this.state.endDate, this.state.startDate));
+  };
+
+  handleChange = (id) => (date) => {
+    this.setState(prev => {
+      prev[id] = date;
+      return prev;
+    }, () => {
+      if (id == 'startDate' || id == 'endDate') {
```

Comment:
This seems brittle, but I don't have a better implementation solution to point you in a better direction. Can you add a comment here that this is tightly coupled to the rendered components?